### PR TITLE
[WIP] document how to gather the audit logs

### PR DESCRIPTION
**Pull request by mburke5678**
https://jira.coreos.com/browse/MSTR-328
https://github.com/openshift/api/pull/279/files
https://bugzilla.redhat.com/show_bug.cgi?id=1695828

### CODE REVIEW

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-audit-log.adoc
+
+[id="nodes-nodes-audit-log-advanced-{context}"]
+= Advanced Audit
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Do we have 2 options for auditing now? Or is it just one? 

If its just one call this 'auditing'!

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+The advanced audit feature provides several improvements over the
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: If it's just auditing, this needs to change, to align

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+basic audit functionality, including fine-grained events filtering and multiple output back ends.
+
+To enable the advanced audit feature, provide the following values in the
+`openshift_master_audit_config` parameter:
+
+----
+openshift_master_audit_config={"enabled": true, "auditFilePath": "/var/lib/origin/oscp-audit.log", "maximumFileRetentionDays": 14, "maximumFileSizeMegabytes": 500, "maximumRetainedFiles": 5, "policyFile": "/etc/origin/master/adv-audit.yaml", "logFormat":"json"}
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Where and on what object do you set this.

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+----
+
+[IMPORTANT]
+====
+The policy file *_/etc/origin/master/adv-audit.yaml_* must be available on each master node.
+====
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Is this true? In 4.x with everything being in a CRD or some other object, what do file paths like this have to do with this configuration?

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+|Specifies the strategy for sending audit events. Allowed values are `block`
+(blocks processing another event until the previous has fully processed) and
+`batch` (buffers events and delivers in batches).
+|===
+
+[IMPORTANT]
+====
+To enable the advanced audit feature, you must provide either `policyFile` *or*
+`policyConfiguration` describing the audit policy rules:
+====
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Don't we provide one by default?

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+.Sample Audit Policy Configuration
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-advanced.adoc]: We should start this section of with a block of text that explains the following as examples of how you can alter the auditing rules and behaviors.

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+[id="nodes-nodes-audit-log-basic-config-{context}"]
+= Configuring the Basic Audit Log
+
+
+To configure the basic audit....
+
+//This is a best guess attempt inspired by https://github.com/openshift/api/pull/279
+
+.Prerequisite
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: So, the basic functionality of auditing, requires something (and operator) that is no longer part of the core product?

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 8]
```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-audit-log.adoc
+
+[id="nodes-nodes-audit-log-basic-config-{context}"]
+= Configuring the Basic Audit Log
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: I'll ask again do we really have 2 audit options in 4.x?

Consider removing this in favor of advanced auditing (renamed to auditing)

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+You can set the log level for both settings to one of the following. The setting can be different for each setting:
++
+* *Normal* Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations. Similar to `glog=2`.
+* *Debug*. Debug is for troubleshooting problems. Even common operations may be logged, and less helpful but more quantity of notices.  Similar to `glog=4`.
+* *Trace*. Trace is for troubleshooting problems when *Debug* is not verbose enough. Logging every function call as part of a common operation, to tracing execution of a query.  Similar to `glog=6`.
+* *TraceAll*. TraceAll is troubleshoting at the level of API content/decoding.  It will dump complete body content. In production clusters, this setting causess performance degradation and results in a significant number of logs.
+Similar to ` glog=8`.
+
+
+//Can we still set these?
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: This question needs to be addressed!

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+|`maximumFileSizeMegabytes`
+|Specifies maximum size in megabytes of the log file before it gets rotated.
+Defaults to 100MB.
+|===
+
+[IMPORTANT]
+====
+Because the {product-title} master API now runs as static pod, you must define
+the `auditFilePath` location in the *_/var/lib/origin_* or *_/etc/origin/master/_* file.
+====
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: I have a comment about this in the advanced section, this seems out of place.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+.Procedure
+
+To view the basic audit log:
+
+. View the {product-title] API server logs
+
+.. If necessary, get the name of the log you want to view:
++
+----
+$ oc --insecure-skip-tls-verify adm node-logs --role=master --path=openshift-apiserver/
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc]: This seems wrong for viewing something from elastic search.

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 12]
```diff
@@ -0,0 +1,33 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-audit-log.adoc
+
+[id="nodes-pods-audit-log-basic-{context}"]
+= About the Basic Audit Log
```
Comment (sferich888) [inline modules/nodes-nodes-audit-log-basic.adoc]: We should clarify if this is basic or advance related. It seems more tied to advanced.

**Review by sferich888 — CHANGES_REQUESTED** [review]
This seems to be out of place with 4.x it needs a deep engineering review.

File: modules/nodes-nodes-audit-log-advanced.adoc (hunk 1 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-advanced.adoc]: @sferich888 This is how the audit feature was presented in 3.x, Basic and Advanced. 
Is there someone who can confirm if the two are available in 4.x?

File: modules/nodes-nodes-audit-log-advanced.adoc (hunk 3 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-advanced.adoc]: @sferich888 I believe it was the _/etc/origin/master/adv-audit.yaml_. Is his file in 4.x?
If so, how do users edit files on OpenShift systems? Previously, the user could SSH in; do they need physical access to edit such files?

File: modules/nodes-nodes-audit-log-basic-viewing.adoc (hunk 11 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-basic-viewing.adoc]: @sferich888 I am not sure this is related to ES. I think this is an API-level log function only. And not related to Elasticsearch(??). 
@jcantrill Can you comment? Is audit logging different from EFK? Is Elasticsearch required in 4.x for audit logging?

**Comment by mburke5678** [comment]
@sferich888 I found where the audit settings are in 4.x, I believe. In a config file on the kube API server. 
I got rid of the Advanced Audit topic for the moment; unless I can find where the advanced settings are. Do you know who would be a good source of info on API auditing in 4.x? 

I am closing this PR in favor of https://github.com/openshift/openshift-docs/pull/14480.

**Pull request closed without merging**